src/dbnode/server/server.go

```diff
@@ -524,6 +521,11 @@ func Run(runOpts RunOptions) {
 		}).
 		SetMmapReporter(mmapReporter).
 		SetQueryLimits(queryLimits)
+
+	if cfg.Index.MaxResultsPerPermit > 0 {
+		indexOpts.SetResultsPerPermit(cfg.Index.MaxResultsPerPermit)
```

Comment:
I believe this should be `indexOpts = indexOpts.SetResultsPerPermit(cfg.Index.MaxResultsPerPermit)` since our options are immutable?